### PR TITLE
Adds Reflector Blobs

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
add: Blobs can now upgrade their strong blobs to reflect projectiles at the cost of their normal health and extra brute resistance.
/:cl:

When creating strong blobs, creating another strong blob on top of another will create a reflective blob. Reflective blobs reflects projectiles based on their ricochet chance, meaning this more or less shuts down laser fire until someone rushes in to break the reflector or uses ballistics on it.

This might be really only useful for stuff like people placing emitters half-way across the station but having more avenues to deal with laserspam isn't bad.

### CODE REVIEW

File: code/modules/antagonists/blob/blob/blobs/shield.dm [hunk 1]
```diff
@@ -3,8 +3,9 @@
 	icon = 'icons/mob/blob.dmi'
 	icon_state = "blob_shield"
 	desc = "A solid wall of slightly twitching tendrils."
+	var/damaged_desc = "A wall of twitching tendrils."
 	max_integrity = 150
-	brute_resist = 0.25
+	brute_resist = 0.2
```
Comment (ChangelingRain) [inline code/modules/antagonists/blob/blob/blobs/shield.dm]: changelog!
this makes shield blobs even harder to damage with brute.

File: code/modules/antagonists/blob/blob/powers.dm [hunk 2]
```diff
@@ -359,7 +369,7 @@
 	to_chat(src, "<b>You can expand, which will attack people, damage objects, or place a Normal Blob if the tile is clear.</b>")
 	to_chat(src, "<i>Normal Blobs</i> will expand your reach and can be upgraded into special blobs that perform certain functions.")
 	to_chat(src, "<b>You can upgrade normal blobs into the following types of blob:</b>")
-	to_chat(src, "<i>Shield Blobs</i> are strong and expensive blobs which take more damage. In additon, they are fireproof and can block air, use these to protect yourself from station fires.")
+	to_chat(src, "<i>Shield Blobs</i> are strong and expensive blobs which take more damage. In additon, they are fireproof and can block air, use these to protect yourself from station fires. Upgrading them again will ressult in a reflective blob, capable of reflecting most projectiles at the cost of the strong blob's extra health.")
```
Comment (ChangelingRain) [inline code/modules/antagonists/blob/blob/powers.dm]: result, not ressult

File: code/modules/antagonists/blob/blob/powers.dm [hunk 3]
```diff
@@ -110,15 +110,25 @@
 		to_chat(src, "<span class='warning'>You now require a nearby node or core to place factory and resource blobs.</span>")
 	else
 		to_chat(src, "<span class='warning'>You no longer require a nearby node or core to place factory and resource blobs.</span>")
-
+	
 /mob/camera/blob/verb/create_shield_power()
 	set category = "Blob"
 	set name = "Create Shield Blob (15)"
```
Comment (ChangelingRain) [inline code/modules/antagonists/blob/blob/powers.dm]: Create/Upgrade

File: code/modules/antagonists/blob/blob/blobs/shield.dm [hunk 4]
```diff
@@ -20,14 +21,37 @@
...
+
+/obj/structure/blob/shield/reflective
+	name = "reflective blob"
+	desc = "A solid wall of slightly twitching tendrils with a reflective glow."
+	damaged_desc = "A wall of twitching tendrils with a reflective glow."
+	icon_state = "blob_glow"
+	flags_1 = CHECK_RICOCHET_1
+	point_return = 8
+	max_integrity = 50
+	brute_resist = 1
```
Comment (ChangelingRain) [inline code/modules/antagonists/blob/blob/blobs/shield.dm]: note that reducing the integrity and brute resistance both means that its eHPvsBrute compared to a normal shield blob is 50, vs 600. Is that how weak these should be versus brute?

Comment (MMMiracles) [reply code/modules/antagonists/blob/blob/blobs/shield.dm]: Lasers have an innate 80% chance to reflect, so this is meant to be a good counter against people spamming lasers at you and nothing else. My ideal scenario is blob places these down and the crew now has to either switch to ballistics to break through or rush up and break it themselves.

File: code/modules/antagonists/blob/blob/blobs/shield.dm [hunk 5]
```diff
@@ -20,14 +21,37 @@
...
 		name = initial(name)
 		desc = initial(desc)
 		atmosblock = TRUE
 	air_update_turf(1)
+
+/obj/structure/blob/shield/reflective
+	name = "reflective blob"
+	desc = "A solid wall of slightly twitching tendrils with a reflective glow."
+	damaged_desc = "A wall of twitching tendrils with a reflective glow."
+	icon_state = "blob_glow"
```
Comment (ChangelingRain) [inline code/modules/antagonists/blob/blob/blobs/shield.dm]: The damaged state for this sprite is different from shield blob's damaged state; here is a free pair of sprites. http://puu.sh/BWf7z/8c78d2aabe.dmi

**Comment by evsey9** [comment]
Why does it take tg THIS LONG to steal good ideas from goon

**Comment by AdamElTablawy** [comment]
Kind of a big investment for a single tile of easily broken blob. Useful in theory, in practice it's not going to be used much for the same reasons that blobs don't usually put strong blobs down in favor of just expanding and spamming nodes and factories.

Could probably just have it be an alternative to strong blobs at the same cost, rather than requiring to spend thirty blob points on one tile.

**Comment by YakumoChen** [comment]
does it reflect randomly or directly back? I think 30 points is a good investment for potentially maiming someone firing a laser at you and forcing crew to stop firing.

**Comment by ExcessiveUseOfCobblestone** [comment]
> 30 resource for a structure that has 1/3 HP and can't even guarantee laser reflection

**Comment by MMMiracles** [comment]
Because these work exactly like titanium walls in how they reflect projectiles, so you're not taking the damage and sending the projectile back into the crowd that fired at you.

**Comment by evsey9** [comment]
Wait, this doesn't have GUARANTEED reflection?
Why bother, then

**Comment by optimumtact** [comment]
It's a lot easier to buff something than nerf it guys

**Pull request merged**